app.rb

```diff
@@ -0,0 +1,187 @@
+require './book'
+require './rental'
+require './student'
+require './teacher'
+
+class App # rubocop:disable Metrics/ClassLength
+  def initialize
+    @people_list = []
+    @books_list = []
+    @rentals_list = []
+  end
+
+  def run_program
+    puts '[a] Create a person'
+    puts '[b] Create a book'
+    puts '[c] Create a rental'
+    puts '[d] List all persons'
+    puts '[e] List all books'
+    puts '[f] List all rentals'
+    puts '[g] Exit'
+    puts
+    print 'Enter the letter corresponding to the operation you want to perform: '
+    user_choice = gets.chomp
+    operator(user_choice)
+  end
+
+  def operator(user_choice)
+    case user_choice
+    when 'a'
+      create_person
+    when 'b'
+      create_book
+    when 'c'
+      create_rental
+    when 'd'
+      list_person
+    when 'e'
+      list_books
+    when 'f'
+      list_rentals
+    else
+      exit
+    end
+  end
+
+  def list_books
+    if @books_list.length.zero?
+      p 'There is no books to show, Add new one'
+    else
+      @books_list.each { |book| puts "Title: #{book.title}, Author: #{book.author}" }
+    end
+    run_program
+  end
+
+  def list_person
+    if @people_list.length.zero?
+      p 'There is no people to show, Add new one'
+    else
+      @people_list.each do |person|
+        p "Person ID: #{person.id}, Name: #{person.name} (#{person.class.name}), Age: #{person.age}"
+        puts
+      end
+    end
+    run_program
+  end
+
+  def create_person
+    p 'What do you want to create?'
+    puts '[1] Student'
+    puts '[2] Teacher'
+    print 'Enter number: '
+    puts
+    choice = gets.chomp
+    case choice
+    when '1' then add_student
+    when '2' then add_teacher
+    else puts 'Please enter a valid input, it must be the number 1 or the number 2'
+         puts
+    end
+    create_person
+  end
+
+  def add_student
+    print 'Student classroom: '
+    classroom = gets.chomp.to_i
+    print 'Student name: '
+    name = gets.chomp
+    print 'Student age: '
+    age = gets.chomp.to_i
+    print 'Has Parent permission? [Y/N]: '
+    input = gets.chomp.downcase
+    valid_permission = input == 'y'
+    @people_list.push(Student.new(classroom, age, name, parent_permission: valid_permission))
+    p 'Student created'
+    puts
+    run_program
+  end
+
+  def add_teacher
+    print 'Teacher age: '
+    age = gets.chomp.to_i
+    print 'Teacher name: '
+    name = gets.chomp
+    print 'Teacher specialization: '
+    specialization = gets.chomp
+    @people_list.push(Teacher.new(age, name, specialization))
+    p 'Teacher created'
+    puts
+    run_program
+  end
```

Comment:
- [ ] I noticed that after creating a teacher when I try to list all persons instead of displaying the name of the teacher it displays the specialization, please kindly fix it as well 🙃 🙃 

<img width="682" alt="Screen Shot 2022-09-02 at 01 54 47" src="https://user-images.githubusercontent.com/74501523/188031978-f4c6b631-fae0-4ad8-8450-c00098b2d355.png">